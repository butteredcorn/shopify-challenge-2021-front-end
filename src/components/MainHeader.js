import {useContext} from 'react'
import styled from 'styled-components'
import { ThemeContext } from "./ThemeProvider"
import { Link } from 'react-router-dom';
import '../styles/MainHeader.css'

const MainHeaderContainer = styled.section`
    background-color: ${props => props.theme.bg.primary};
    color: ${props => props.theme.text.tertiary};
    border: 1px solid ${props => props.theme.bg.primary};
`

const MainHeader = () => {
    const { currentTheme, dispatch } = useContext(ThemeContext);

    const toggleTheme = () => {
        dispatch({ type: "toggleTheme" });
    };

    return (
        <MainHeaderContainer>
            <header className="home-header">
                <h1><Link to="/" className="nav-link">The Shoppies</Link></h1>
                <p className="toggle" onClick={toggleTheme}>{currentTheme.id} Theme</p>
            </header>
        </MainHeaderContainer>
    );
  };
  export default MainHeader;