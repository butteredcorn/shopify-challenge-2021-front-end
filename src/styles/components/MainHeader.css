.toggle {
    text-transform: capitalize;
}

.home-header {
    min-height: 10vh;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    font-size: 2.5rem;
    font-weight: bold;
    margin: 1rem;
  }

.nav-link {
    text-decoration: none;
}