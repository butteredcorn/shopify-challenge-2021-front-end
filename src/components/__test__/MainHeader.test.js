import { render, screen, cleanup, fireEvent, getByText } from '@testing-library/react';
import renderer from 'react-test-renderer'
import {BrowserRouter} from 'react-router-dom'
import ThemesProvider from '../../providers/ThemeProvider'
import MainHeader from '../MainHeader';

afterEach(cleanup);

it('renders MainHeader component with title "The Shoppies"', () => {
  render(<BrowserRouter><ThemesProvider><MainHeader/></ThemesProvider></BrowserRouter>);
  expect(screen.getByText(/The Shoppies/i)).toBeInTheDocument();
});

it('renders MainHeader with the "Light Theme"', () => {
    render(<BrowserRouter><ThemesProvider><MainHeader/></ThemesProvider></BrowserRouter>);
    expect(screen.getByText(/Light Theme/i)).toBeInTheDocument();
});

it('renders MainHeader with "Dark Theme" when toggled', () => {
    const {container} = render(<BrowserRouter><ThemesProvider><MainHeader/></ThemesProvider></BrowserRouter>);
    fireEvent(
        getByText(container, 'light Theme'),
        new MouseEvent('click', {
            bubbles: true,
            cancelable: true
        })
    )
    expect(screen.getByText(/Dark Theme/i)).toBeInTheDocument();
});

it('matches a snapshot', () => {
  const json = renderer.create(<BrowserRouter><ThemesProvider><MainHeader/></ThemesProvider></BrowserRouter>).toJSON();
  expect(json).toMatchSnapshot();
});