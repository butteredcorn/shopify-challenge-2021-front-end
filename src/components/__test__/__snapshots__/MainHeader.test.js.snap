// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<section
  className="sc-bdvvaa cuEABP"
>
  <header
    className="home-header"
  >
    <h1>
      <a
        className="nav-link"
        href="/"
        onClick={[Function]}
      >
        The Shoppies
      </a>
    </h1>
    <p
      className="toggle"
      onClick={[Function]}
    >
      light
       Theme
    </p>
  </header>
</section>
`;
